src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1061,12 +1060,24 @@ void SotaUptaneClient::sendDeviceData() {
 
 result::UpdateCheck SotaUptaneClient::fetchMeta() {
   result::UpdateCheck result;
+
+  reportNetworkInfo();
+
+  if (hasPendingUpdates()) {
+    // no need in update checking if there are some pending updates
+    return result::UpdateCheck({}, 0, result::UpdateStatus::kError, Json::nullValue,
+                               "There are pending updates, no new updates are checked");
+  }
+
   if (updateMeta()) {
+    // is updateMeta() part of the update checking procedure ?
+    // if no then the sendEvent should be at the end of this block
```

Comment:
Yes, `updateMeta` is an essential part of the update checking process. In fact, arguably `fetchMeta` is a bit of a misnomer. The issue is another legacy concept: we used to distinguish between fetching the metadata and then actually checking it for updates. Now we generally do it all at once. I think the program flow you have here is probably exactly what is desired.

Reply:
ok